api/v1beta1/nutanix_types.go

```diff
@@ -71,3 +80,18 @@ type NutanixCategoryIdentifier struct {
 	// +optional
 	Value string `json:"value,omitempty"`
 }
+
+type NutanixGPU struct {
+	// Type is the identifier type to use for this resource.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=deviceId;name
+	Type NutanixGPUIdentifierType `json:"type"`
+
+	// deviceID is the id of the GPU entity.
+	// +optional
+	DeviceId *int64 `json:"deviceId,omitempty"`
```

Comment:
nit: can the json property be `device_id` instead?

Reply:
We don't use _ in any other property. Will rename to deviceID instead.